### PR TITLE
chore: Hard deprecate Drawer component

### PR DESCRIPTION
**Pull request by RayRedGoose**
## Summary

Resolves: #1931 

Drawer has been soft-deprecated in v8 and can be removed in v9.

###  Acceptance Criteria
- [x] The component removed
- [x] The component story removed
- [x] The component tests removed
- [x] The upgrade guide updated

## Release Category
Components

### BREAKING CHANGES
The `Drawer` component was hard deprecated and fully removed from Canvas Kit. Please replace the deprecated `Drawer` component with `SidePanel` from the Preview package.

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [standard MDX template](https://github.com/Workday/canvas-kit/discussions/1131)

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)
- [x] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

### CODE REVIEW

File: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import {Meta} from '@storybook/addon-docs';
+
+<Meta title="Welcome/Upgrade Guides/v8.0" />
+
+# Canvas Kit 9.0 Upgrade Guide
+
+This guide contains an overview of the changes in Canvas Kit v8. Please
```
Comment (mannycarrera4) [inline modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:7]:
```suggestion
This guide contains an overview of the changes in Canvas Kit v9. Please
```

**Comment by mannycarrera4** [comment]
I think we need to highlight or soft deprecate side panel in main, because i can see people consuming the side panel in main and I don't think that's what we want

File: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import {Meta} from '@storybook/addon-docs';
+
+<Meta title="Welcome/Upgrade Guides/v8.0" />
```
Comment (jaclynjessup) [inline modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:3]:
```suggestion
<Meta title="Welcome/Upgrade Guides/v9.0" />
```

**Comment by jaclynjessup** [comment]
@mannycarrera4 in regards to your comment about Side Panel in main - we do have #1316 open - we can discuss if this is a must have / should have for v9

File: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+prescribed in our Upgrade Guide to update your application. Please
+[reach out](https://github.com/Workday/canvas-kit/issues/new?labels=bug&template=bug.md) to our team
+directly if you need additional help.
+
+---
+
+### Drawer
+
+We've [hard-deprecated](#hard-deprecation) `Drawer` and this component was fully removed from Canvas
+Kit. Consider using the [Side Panel](/components/containers/side-panel/). Aware that `SidePanel`
```
Comment (mannycarrera4) [inline modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:71]: i think this links to the one in main, i think this should be the one from preview right?

Comment (RayRedGoose) [reply modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:71]: Yeah, I can change it to preview one, but we probably will need to change it after SidePanel deprecation

Comment (jaclynjessup) [reply modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:71]: should we leave a comment as a TODO so we ensure we review where we are pointing users before we release v9?

File: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### Drawer
+
+<!---
+TODO: Please review this section with documenting SidePanel deprication
+-->
+
+We've [hard-deprecated](#hard-deprecation) `Drawer` and this component was fully removed from Canvas
+Kit. Consider using the
+[Side Panel](https://workday.github.io/canvas-kit/?path=/story/preview-side-panel--basic). Aware
```
Comment (mannycarrera4) [inline modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:76]:
```suggestion
[Side Panel](https://workday.github.io/canvas-kit/?path=/story/preview-side-panel--basic). Be Aware
```

File: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+that `SidePanel` from main package is planned to be deprecated and replaced by one from Preview. So
```
Comment (mannycarrera4) [inline modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:77]:
```suggestion
that `SidePanel` from the main package is planned to be replaced by the one from Preview. So
```

File: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+we suggest to use preview version of `SidePanel` until main version `SidePanel` hard deprecation.
```
Comment (mannycarrera4) [inline modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:78]:
```suggestion
We suggest you use the Preview version of `SidePanel` until the one in Main is replaced with it.
```

File: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx [hunk 7]
```diff
@@ -0,0 +1,80 @@
+import {Meta} from '@storybook/addon-docs';
+
+<Meta title="Welcome/Upgrade Guides/v9.0" />
```
Comment (mannycarrera4) [inline modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:3]:
```suggestion
<Meta title="Guides/Upgrade Guides/v9.0" />
```

File: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+---
+
+### Drawer
+
+<!---
+TODO: Please review this section with documenting SidePanel deprication
+-->
+
+We've [hard-deprecated](#hard-deprecation) `Drawer` and this component was fully removed from Canvas
```
Comment (mannycarrera4) [inline modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:74]:
```suggestion
We've [hard-deprecated](#hard-deprecation) `Drawer` and this component has been fully removed from Canvas
```

File: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+Kit. Consider using the
+[Side Panel](https://workday.github.io/canvas-kit/?path=/story/preview-side-panel--basic). Be Aware
+that `SidePanel` from the main package is planned to be replaced by the one from Preview.
```
Comment (mannycarrera4) [inline modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx:77]:
```suggestion
that `SidePanel` from the Main package is planned to be replaced by the one from Preview.
```

**Review by mannycarrera4 — APPROVED** [review]

**Pull request merged**